### PR TITLE
Make example generation work when the model is torch.compile-d

### PR DESCRIPTION
**Pull request by IggShaman**
Andrej mentions this issue a couple of times in the video. Apparently, when a model is compiled, it will fix its input and output tensor sizes. The training regime works with inputs of shape (B, T), currently configured to (64, 1024). Example generation works with input tensors of shapes (4, 8) ... (4, 32).
A simple workaround here is to pad example generation tensors to the (B, T) shape, and ignore extra rows. The same workaround applies to the hellaswag eval.
Unfortunately, this increases the computation time linearly to both the B and T, so quadratically overall.

### CODE REVIEW

**Comment by jasonrichdarmawan** [comment]
Any reason you didn't fix the HellaSwag as well? I haven't rent 8x A100 yet so I am not sure whether this works or not. But I think the code below would work?

```
...
      # render the example into tokens and labels
      _, tokens, mask, label = render_example(example)
      tokens = tokens.to(device)
      mask = mask.to(device)

      pad_x_to = (T if use_compile else tokens.size(1)) - tokens.size(1)
      tokens = torch.cat((tokens, torch.zeros([tokens.size(0), pad_x_to], dtype=torch.long)), dim=1)
      mask = torch.cat((mask, torch.zeros([mask.size(0), pad_x_to], dtype=torch.long)), dim=1)
```

**Comment by hurrypeter02-cmd** [comment]
Perhaps we can avoid repeated recompilation by wrapping it in a generate‑style function.

` @torch.no_grad()
    def generate(self,token,max_len):
        assert token.ndim==1
        token = token.unsqueeze(0)
        while(True):
            if token.shape[-1] < max_len:
                token_context = token[:,-self.config.block_size:]
                logits,loss = self(token_context)
                logits = logits[:,-1,:config.vocab_size]
                pro = F.softmax(logits,dim=-1)
                topk_pro,topk_indices = torch.topk(pro,50,dim=-1)
                ix = torch.multinomial(topk_pro ,num_samples=1)
                next_idx = torch.gather(topk_indices,-1,ix)
                token = torch.cat([token,next_idx],dim=-1)
            else:
                break;
        token = token.squeeze(0)
        return token
`

`if master_process:
            raw_model.eval()
            prompt = "Hi,i am a language model,"
            enc = tiktoken.get_encoding("gpt2")
            tokens = enc.encode(prompt)
            tokens = torch.tensor(tokens,dtype=torch.long,device=device)
            generated = raw_model.generate(tokens,max_len=tokens.numel()+80)
            prompts = enc.decode(generated.tolist())
            print(prompts)
            raw_model.train()`

[https://github.com/[hurrypeter02-cmd/gpt-from-scratch](https://github.com/hurrypeter02-cmd/gpt-from-scratch/tree/main)/tree/main](url)
